x-pack/plugins/ingest_manager/server/routes/data_streams/handlers.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { RequestHandler } from 'src/core/server';
+import { DataStream } from '../../types';
+import { GetDataStreamsResponse } from '../../../common';
+
+const DATA_STREAM_INDEX_PATTERN = 'logs-*-*,metrics-*-*';
+
+export const getListHandler: RequestHandler = async (context, request, response) => {
+  const callCluster = context.core.elasticsearch.dataClient.callAsCurrentUser;
+
+  try {
+    // Get stats (size on disk) of all potentially matching indices
+    const { indices: indexStats } = await callCluster('indices.stats', {
+      index: DATA_STREAM_INDEX_PATTERN,
+      metric: ['store'],
+    });
+
+    // Get all matching indices and info about each
+    // This returns the top 100,000 indices (as buckets) by last activity
+    const {
+      aggregations: {
+        index: { buckets: indexResults },
+      },
+    } = await callCluster('search', {
+      index: DATA_STREAM_INDEX_PATTERN,
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            must: [
+              {
+                exists: {
+                  field: 'fields.stream.namespace',
```

Comment:
That it has a `fields` prefix is actually a bug. As soon as https://github.com/elastic/beats/pull/17858 is merged, this must be adjusted. Same for the other fields below.

@michalpristas Seeing your PR is approved, could you get it in so perhaps we can adjust this here before merging?

Reply:
Will adjust.